src/Microsoft.XmlSerializer.Generator/src/Sgen.cs

```diff
@@ -232,23 +232,37 @@ private void GenerateFile(List<string> typeNames, string assemblyName, bool prox
             {
                 Type type = types[i];
 
-                if (type != null)
+                try
                 {
-                    bool isObsolete = false;
-                    object[] obsoleteAttributes = type.GetCustomAttributes(typeof(ObsoleteAttribute), false);
-                    foreach (object attribute in obsoleteAttributes)
+                    if (type != null)
                     {
-                        if (((ObsoleteAttribute)attribute).IsError)
+                        bool isObsolete = false;
+                        object[] obsoleteAttributes = type.GetCustomAttributes(typeof(ObsoleteAttribute), false);
+                        foreach (object attribute in obsoleteAttributes)
                         {
-                            isObsolete = true;
-                            break;
+                            if (((ObsoleteAttribute)attribute).IsError)
+                            {
+                                isObsolete = true;
+                                break;
+                            }
                         }
-                    }
 
-                    if (isObsolete)
+                        if (isObsolete)
+                        {
+                            continue;
+                        }
+                    }
+                }
+                //Ignore the FileNotFoundException for any type when call GetCustomAttributes
```

Comment:
Can we make the comment to explain what is the scenario the FNFE can be thrown here? It will be more useful in the future.

Reply:
@zhenlan How about "Ignore the FileNotFoundException when call GetCustomAttributes e.g. if the type uses the attributes defined in a different assembly"